models/emoe_v1.py

```diff
@@ -0,0 +1,1480 @@
+# coding=utf-8
+#
+# Copyright Andrew Greene and Bobak Hashemi, 2024. All rights reserved.
+# This code was modified from the Huggingface Transformers library.
+# https://github.com/huggingface/transformers/blob/v4.37.2/src/transformers/models/llama/modeling_llama.py
+#
+# Copyright 2022 EleutherAI, the HuggingFace Inc. team. All rights reserved.
+#
+# This code is based on EleutherAI's GPT-NeoX library and the GPT-NeoX
+# and OPT implementations in this library. It has been modified from its
+# original forms to accommodate minor architectural differences compared
+# to GPT-NeoX and OPT used by the Meta AI team that trained the model.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Elastic Sparse LLaMA model."""
```

Comment:
Do we really need this? IIUC your comment, we just do `model1 = LlamaPretrainedModel(...); model2 = LlamaPretrainedModel(...)` We don't need to import from a diff file.

Reply:
we don't need this; I'd thought that being able to e.g.
from emoe_v1 import ModelV1
would make it easier to iterate on, but it's not required by any means